voidcrew/code/__HELPERS/game.dm

```diff
@@ -19,3 +19,20 @@
 			to_chat(character, "<h2>Tag your ship with KOS to fight NT-C, SYN-C, and KOS ships</h2>")
 		else
 			return
+
+/proc/AnnounceArrival(mob/living/carbon/human/character, rank, obj/structure/overmap/ship/simulated/ship)
+	if(!SSticker.IsRoundInProgress() || QDELETED(character))
+		return
+	var/area/A = get_area(character)
+	deadchat_broadcast("<span class='game'> has arrived on the <span class='name'>[ship.name]</span> at <span class='name'>[A.name]</span>.</span>", "<span class='game'><span class='name'>[character.real_name]</span> ([rank])</span>", follow_target = character, message_type=DEADCHAT_ARRIVALRATTLE)
+	if((!GLOB.announcement_systems.len) || (!character.mind))
+		return
+	if((character.mind.assigned_role == "Cyborg") || (character.mind.assigned_role == character.mind.special_role))
+		return
+
+	//Alt job titles
+	var/displayed_rank = rank
+	if(character.client && character.client.prefs && character.client.prefs.alt_titles_preferences[rank])
```

Comment:
are we gonna keep alt titles in the rebase? If not we might as well not port it to voidcrew code.
I don't think alt titles even *work* since jobs have unique names per ship.

Reply:
good idea